### PR TITLE
[Tests] Update MT4134 tests for the new Xcode 11.4

### PR DESCRIPTION
**Pull request by mandel-macaque**
New frameworks were added and the test needs to be updated.

fixes: https://github.com/xamarin/maccore/issues/954

### CODE REVIEW

**Comment by spouliot** [comment]
How is that related to `[FAIL] NSSet1Test.IEnumerableTest` failing ?

File: tests/mtouch/RegistrarTest.cs [hunk 1]
```diff
@@ -334,6 +334,14 @@ public void MT4134 ()
...
 					new { Framework = "IdentityLookupUI", Version = "12.0" },
 					new { Framework = "NaturalLanguage", Version = "12.0" },
+					new { Framework = "VisionKit", Version = "13.0" },
+					new { Framework = "SoundAnalysis", Version = "13.0" },
+					new { Framework = "PencilKit", Version = "13.0" },
+					new { Framework = "MetricKit", Version = "13.0" },
+					new { Framework = "LinkPresentation", Version = "13.0" },
+					new { Framework = "CoreHaptics", Version = "13.0" },
+					new { Framework = "BackgroundTasks", Version = "13.0" },
+					new { Framework = "QuickLookThumbnailing", Version = "13.0" },
```
Comment (spouliot) [inline tests/mtouch/RegistrarTest.cs]: Looks good - but did this work in `xcode11`, `xcode11.1`, `xcode11.2` and `xcode11.3` branches ?

Comment (mandel-macaque) [reply tests/mtouch/RegistrarTest.cs]: That is a very good question, I just saw the issue starting, but I did not check on the old ones, I'll take a look. Did you see it in the last monitoring spring?

Comment (spouliot) [reply tests/mtouch/RegistrarTest.cs]: no, that's why I'm surprised
maybe device only ?

Comment (spouliot) [reply tests/mtouch/RegistrarTest.cs]: hmm... even if device only those all branches were merged to master so we should have seen failures there (at least)

Comment (mandel-macaque) [reply tests/mtouch/RegistrarTest.cs]: Here it is failing: https://github.com/xamarin/xamarin-macios/commit/c3001c8331261e4b331084dea498623ef1d2e6fa

Test results: http://xamarin-storage/jenkins/xamarin-macios/xcode11.4/c3001c8331261e4b331084dea498623ef1d2e6fa/4/jenkins-results/tests/index.html

MTouch tests:
```
Xamarin.Registrar.MT4134: errors
Expected: 15
But was: 23
```

So I'm surprised too that it did not happen in the previous spring :/

**Comment by mandel-macaque** [comment]
@spouliot is there an example of that? Or just an update me?

**Comment by spouliot** [comment]
@mandel-macaque just add something around https://github.com/xamarin/xamarin-macios/blob/bc878256f4a0e6edf52469423a8cd27a378c61f2/tools/common/Frameworks.cs#L327

**Comment by mandel-macaque** [comment]
Known issue: https://github.com/xamarin/xamarin-macios/pull/7936

File: tests/mtouch/RegistrarTest.cs [hunk 2]
```diff
@@ -317,7 +317,8 @@ public void MT4134 ()
 				mtouch.CreateTemporaryApp (extraCode: sb.ToString (), usings: "using System; using Foundation; using ObjCRuntime;", extraArg: "/debug:full");
 				mtouch.Linker = MTouchLinker.LinkSdk;
 				mtouch.Registrar = MTouchRegistrar.Static;
-				mtouch.AssertExecuteFailure (MTouchAction.BuildDev);
+				mtouch.AssertExecuteFailure (MTouchAction.BuildDev);BackgroundTasks
```
Comment (spouliot) [inline tests/mtouch/RegistrarTest.cs]: ^ that won't compile ;-)

File: tools/common/Frameworks.cs [hunk 3]
```diff
@@ -327,6 +327,10 @@ public static Frameworks GetiOSFrameworks (Application app)
 				// the above MUST be kept in sync with simlauncher
 				// see tools/mtouch/Makefile
 				// please also keep it sorted to ease comparison
+        // 
+        // The following tests also need to be updated:
+        // 
+        // * RegistrarTest.MT4134
```
Comment (spouliot) [inline tools/common/Frameworks.cs]: alignment ^^

File: tests/mtouch/RegistrarTest.cs (hunk 2 above)
Comment (mandel-macaque) [reply tests/mtouch/RegistrarTest.cs]: stupid manuel

**Comment by mandel-macaque** [comment]
Known issue: https://github.com/xamarin/maccore/issues/2128

File: tests/mtouch/RegistrarTest.cs (hunk 1 above)
Comment (rolfbjarne) [reply tests/mtouch/RegistrarTest.cs]: This is the reason it happens only rarely: 

https://github.com/xamarin/xamarin-macios/pull/7931/files#diff-d231584bddcb738d5232e288f51ae8d6R288-R290

It requires Xcode 8, which isn't installed in most places.

**Pull request merged**